docs/adr-572-validate-config-for-doctor.md

```
@@ -0,0 +1,85 @@
+# ADR: Use Idiomatic `Option<&PathBuf>` for `validate_config_for_doctor`
+
+## Status
```

markdownlint-cli2 (0.22.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)